src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -275,11 +287,11 @@ private static string SymbolFilePath
 
         internal static bool CategoryExists(string machine, string category)
         {
-            PerformanceCounterLib library = GetPerformanceCounterLib(machine, new CultureInfo(EnglishLCID));
+            PerformanceCounterLib library = GetPerformanceCounterLib(machine, EnglishCulture);
             if (library.CategoryExists(category))
                 return true;
 
-            if (CultureInfo.CurrentCulture.Parent.LCID != EnglishLCID)
+            if (CultureInfo.CurrentCulture.Parent.Name != EnglishCulture.Name)
```

Comment:
For my own education, do you mind explaining why we don't want to compare using LCID?

I am looking at the [MS Docs code example](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.lcid?view=net-6.0) for that property and what I'm understanding is that "Name" is the general category, and LCID is the specific sub-category.

Alternatively, why isn't `CultureInfo.Equals` a good choice for comparison?

Reply:
When running with Globalization Invariant mode, we don't support the LCID functionality at all. This means you cannot create a culture using the LCID. And if you request LCID for any culture (except Invariant culture) it will get a constant value like 0x1000.

> Alternatively, why isn't CultureInfo.Equals a good choice for comparison?

CultureInfo doesn't override the object Equals method. This means this comparison will be a reference equality and not content values equality. You can have 2 culture instances with the same name and Equals return false when comparing them.

Reply:
OK got it. Thanks.